modules/account-lib/src/coin/stx/transactionBuilder.ts

```diff
@@ -0,0 +1,251 @@
+      }
+    });
+  }
+
+  /**
+   * Set the transaction fees
+   *
+   * @param {BaseFee} fee The maximum gas to pay
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  fee(fee: BaseFee): this {
+    this.validateValue(new BigNumber(fee.fee));
+    this._fee = fee;
+    return this;
+  }
+
+  network(stacksNetwork: StacksNetwork): this {
+    this._network = stacksNetwork;
+    return this;
+  }
+
+  nonce(n: number): this {
+    this._nonce = n;
+    return this;
+  }
+
+  /**
+   *  Set the memo
+   *
+   * @param {string} memo
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  memo(memo: string): this {
+    this._memo = memo;
+    return this;
+  }
+
+  /**
+   *  Set the number of signatures for multi-sig
+   *
+   * @param {number} numSigns
+   * @returns {TransactionBuilder} This transaction builder
+   */
+  numberSignatures(numSigns: number): this {
+    this.validateValue(new BigNumber(numSigns));
+    this._numberSignatures = numSigns;
+    return this;
+  }
+
+  // region Validators
+  /** @inheritdoc */
+  validateAddress(address: BaseAddress, addressFormat?: string): void {
+    if (!isValidAddress(address.address)) {
+      throw new BuildTransactionError('Invalid address ' + address.address);
+    }
+  }
+
+  /** @inheritdoc */
+  validateKey(key: BaseKey): void {
+    const keyPair = new KeyPair({ prv: key.key });
+    if (!keyPair.getKeys().prv) {
+      throw new BuildTransactionError('Invalid key');
+    }
+  }
+
+  /** @inheritdoc */
+  validateRawTransaction(rawTransaction: string): void {
+    if (!rawTransaction) {
+      throw new InvalidTransactionError('Raw transaction is empty');
+    }
+    try {
+      deserializeTransaction(BufferReader.fromBuffer(Buffer.from(removeHexPrefix(rawTransaction), 'hex')));
+    } catch (e) {
+      throw new ParseTransactionError('There was an error parsing the raw transaction');
+    }
+  }
+
+  /** @inheritdoc */
+  validateTransaction(transaction?: Transaction): void {
+    this.validateFee();
+    this.validateNonce();
+  }
+
+  /** @inheritdoc */
+  validateValue(value: BigNumber): void {
+    if (value.isLessThan(0)) {
+      throw new BuildTransactionError('Value cannot be less than zero');
+    }
+  }
+
+  /**
+   * Validates that the fee field is defined
+   */
+  private validateFee(): void {
+    if (this._fee === undefined) {
+      throw new BuildTransactionError('Invalid transaction: missing fee');
+    }
+    try {
+      this.validateValue(new BigNumber(this._fee.fee));
+    } catch (e) {
+      throw new BuildTransactionError('Invalid fee');
+    }
+  }
+
+  /**
+   * Validates that nonce is defined
+   */
+  private validateNonce(): void {
+    if (this._nonce === undefined) {
+      throw new BuildTransactionError('Invalid transaction: missing nonce');
+    }
+    try {
+      this.validateValue(new BigNumber(this._nonce));
+    } catch (e) {
+      throw new BuildTransactionError('Invalid nonce');
```

Comment:
It may be useful to include the nonce value here so we have some more info if we need to debug bad nonces.